### PR TITLE
Use Plugin Framework bridge to bring in new resources

### PR DESCRIPTION
**Pull request by t0yv0**
Fixes https://github.com/pulumi/pulumi-aws/issues/2548

This does a couple of things on top of rquitales/upgrade-v5-alpha.1:

- upgrades the bridge to the latest
- installs PF bridge to serve resources off Plugin Framework directly
- adds some tests that lightly verify some of the new PF-based resources

### CODE REVIEW

File: provider/resources.go [hunk 1]
```diff
@@ -6100,7 +6128,6 @@ func Provider() tfbridge.ProviderInfo {
 			"aws_security_group":                      {Tok: awsDataSource(ec2Mod, "getSecurityGroup")},
 			"aws_security_groups":                     {Tok: awsDataSource(ec2Mod, "getSecurityGroups")},
 			"aws_subnet":                              {Tok: awsDataSource(ec2Mod, "getSubnet")},
-			"aws_subnet_ids":                          {Tok: awsDataSource(ec2Mod, "getSubnetIds")},
```
Comment (t0yv0) [inline provider/resources.go:6103]: Probably orthogonal to this PR, it resolves a warning:

```
warning: Pulumi token "aws:ec2/getSubnetIds:getSubnetIds" is mapped to TF provider data source "aws_subnet_ids", but no such data source found. The mapping will be ignored in the generated provider
```

From ./upstream:

```
.changelog/22743.txt:data-source/aws_subnet_ids: The `aws_subnet_ids` data source has been deprecated and will be removed in a future version. Use the `aws_subnets` data source instead
```

File: provider/resources.go [hunk 2]
```diff
@@ -6986,16 +7013,96 @@ func Provider() tfbridge.ProviderInfo {
...
-	contract.AssertNoErrorf(err, "failed to apply default token mappings")
-
 	prov.SetAutonaming(255, "-")
 
 	// Add a CSharp-specific override for aws_s3_bucket.bucket.
 	prov.Resources["aws_s3_bucket_legacy"].Fields["bucket"].CSharpName = "BucketName"
 
-	return prov
+	pluginFrameworkResoures := map[string]*tfbridge.ResourceInfo{
+		"aws_auditmanager_account_registration": {
```
Comment (t0yv0) [inline provider/resources.go:7022]: These new resources come from PF.

**Comment by t0yv0** [comment]
This did not run tests. Changing base to master to try to make the tests run.

**Comment by t0yv0** [comment]
I think prerequisites compilation should unblock once https://github.com/pulumi/pulumi-aws/pull/2558 is in.

**Comment by kpitzen** [comment]
Looks like there's still some `v5` -> `v6` changes required to resolve the clean working tree issue here.

File: examples/go.mod [hunk 3]
```diff
@@ -3,138 +3,191 @@ module github.com/pulumi/pulumi-aws/examples/v6
 go 1.18
 
 require (
-	github.com/aws/aws-sdk-go v1.38.35
-	github.com/pulumi/pulumi/pkg/v3 v3.31.1
-	github.com/stretchr/testify v1.7.1
+	github.com/aws/aws-sdk-go v1.44.122
+	github.com/pulumi/pulumi/pkg/v3 v3.71.0
```
Comment (t0yv0) [inline examples/go.mod:7]: This implicitly upgrades ProgramTest framework hitting https://github.com/pulumi/pulumi/issues/12859 ; I think this explains TestAccWebServer failure.

**Review by rquitales — APPROVED** [review]
The PR seems to work as intended. A dependent commit I have on top of this to enable more resources also works.

**Pull request merged**